### PR TITLE
Images layers extension unit tests

### PR DESCRIPTION
**Pull request by feloy**
Add unit tests

### CODE REVIEW

**Review by jeffmaury — APPROVED** [review]
LGTM but seems switch to Node 20 is unrelated ?

File: src/explorer.spec.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
+import { expect, test, vi } from "vitest";
```
Comment (benoitf) [inline src/explorer.spec.ts:1]: missing copyright headers (usually it should be raised by a linter rule ?)

File: src/explorer.spec.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+import { Explorer } from "./explorer";
```
Comment (benoitf) [inline src/explorer.spec.ts:2]: should use single quotes (linter is not working ? )

File: src/explorer.spec.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+import { containerEngine } from '@podman-desktop/api';
+import { cp } from "fs/promises";
```
Comment (benoitf) [inline src/explorer.spec.ts:4]: should use node: prefix (linter ?)

File: src/explorer.spec.ts [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+import * as path from "path";
```
Comment (benoitf) [inline src/explorer.spec.ts:5]: should use node: prefix (linter ?)

File: src/explorer.ts [hunk 5]
```diff
@@ -0,0 +1,130 @@
+import * as extensionApi from '@podman-desktop/api';
```
Comment (benoitf) [inline src/explorer.ts:1]: copyright headers

File: src/explorer.ts [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+  empty_layer?: boolean;
+}
+
+export class Explorer {
+
+  #provider: extensionApi.ImageFilesProvider | undefined;
+
+  setProvider(provider: extensionApi.ImageFilesProvider) {
+    this.#provider = provider;
+  }
```
Comment (benoitf) [inline src/explorer.ts:21]: should it be a constructor thingy ? rather than a setter

Comment (feloy) [reply src/explorer.ts:38]: I would prefer, but I don't think it is possible, as we have to build the Explorer before we have the `provider` value:

```
const explorer = new Explorer();
provider = extensionApi.provider.createImageFilesProvider({
  getFilesystemLayers: explorer.getFilesystemLayers.bind(explorer),
});
explorer.setProvider(provider);
```

**Pull request merged**